_posts/2022-01-10-reassign-partitions.md

```diff
@@ -0,0 +1,363 @@
+
+Proposed partition reassignment configuration
+{"version":1,"partitions":[{"topic":"my-topic-1","partition":0,"replicas":[0,1,2],"log_dirs":["any","any","any"]},{"topic":"my-topic-1","partition":1,"replicas":[1,2,0],"log_dirs":["any","any","any"]},{"topic":"my-topic-1","partition":2,"replicas":[2,0,1],"log_dirs":["any","any","any"]},{"topic":"my-topic-1","partition":3,"replicas":[0,2,1],"log_dirs":["any","any","any"]},{"topic":"my-topic-1","partition":4,"replicas":[1,0,2],"log_dirs":["any","any","any"]},{"topic":"my-topic-1","partition":5,"replicas":[2,1,0],"log_dirs":["any","any","any"]},{"topic":"my-topic-1","partition":6,"replicas":[0,1,2],"log_dirs":["any","any","any"]},{"topic":"my-topic-1","partition":7,"replicas":[1,2,0],"log_dirs":["any","any","any"]},{"topic":"my-topic-1","partition":8,"replicas":[2,0,1],"log_dirs":["any","any","any"]},{"topic":"my-topic-1","partition":9,"replicas":[0,2,1],"log_dirs":["any","any","any"]}]}
+```
+Now you can create a `reassignment.json` file and copy this proposed `reassignment.json` data to use in the `reassignment.json` file. You can also create/alter the `reassignment.json` data in the file.
+
+### Scaling down the cluster
+
+Coming to the last step of this example, Here we will see how we use the generated `reassignment.json` to get the reassignment done by the reassignment partition tool.
+
+Since we have copied the `reassignment.json` data into the `reassignment.json` file in the previous step, now we can use it for our next step.
+
+Let's copy the `reassignment.json` file to the interactive pod container.
+
+```sh
+kubectl cp reassignment.json <INTERACTIVE-POD-NAME>:/tmp/reassignment.json
+```
+
+Now we'll start a shell process inside the interactive pod container to run our Kafka bin script.
+```sh
+kubectl exec -ti <INTERACTIVE-POD-NAME> /bin/bash
+```
+
+So let's run the `kafka-reassign-partitions.sh` script now to start the partition reassignment 
+
+```sh
+bin/kafka-reassign-partitions.sh --bootstrap-server <CLUSTER-NAME>-kafka-bootstrap:9092 \
+ --reassignment-json-file /tmp/reassignment.json \ 
+ --execute
+```
+
+You can use the `--verify` mode to check if the partition reassignment is done or if it is still running.
+
+```sh
+bin/kafka-reassign-partitions.sh --bootstrap-server <CLUSTER-NAME>-kafka-bootstrap:9092 \
```

Comment:
It's worth mentioning users may need to run this command multiple times as a reassignment may take a while to complete.